src/client/components/modals/Organization.js

```diff
@@ -0,0 +1,244 @@
+import React, { useContext, useState } from 'react'
+import { Field, Text, Button, theme } from '@aragon/ui'
+import PropTypes from 'prop-types'
+import { ModalWrapper, DisplayErrors } from './ModalWrapper'
+import { validateDAOAddress } from '../../utils/validation'
+import { BoxContext } from '../../wrappers/box'
+import { ModalContext } from '../../wrappers/modal'
+import { close } from '../../stateManagers/modal'
+import { isEmpty } from 'lodash'
+import styled from 'styled-components'
+import {
+  requestedCheckMembership,
+  requestedCheckMembershipClean,
+  requestedCheckMembershipSuccess,
+  requestedCheckMembershipError,
+} from '../../stateManagers/box'
+import {
+  TextInputWithValidation,
+  AnimationLoadingCircle,
+  AddOrganizationSuccess,
+  AddOrganizationError,
+} from '../../components/styled-components'
+
+const Organization = ({
+  ethereumAddress,
+  getFormValue,
+  onChange,
+  saveProfile,
+  organizationId,
+  savingError,
+}) => {
+  const [validationErrors, setValidationErrors] = useState({})
+  const { boxes, dispatch } = useContext(BoxContext)
+  const { dispatchModal } = useContext(ModalContext)
+
+  let {
```

Comment:
Ok... even in case of switches like checkedMembershipSuccess? I didn't use useProfile, because no other manipulation of profile (like saving), is done through the hook. The hook knows about boxes[ethereumAddress].publicProfile, and boxes[ethereumAddress].* would be more like new functionality, more than simple extension of existing one. Should I add it?

Reply:
Good point here. What do you think about creating a new hook called `useProfileState` that allows components to hook into all those stateful variables?